### PR TITLE
Health check endpoints

### PR DESCRIPTION
**Pull request by BenBJSS**

### CODE REVIEW

File: app/apiProvider.js [hunk 1]
```diff
@@ -8,6 +8,18 @@ export class ApiProvider {
     this.documentHost = documentHost;
   }
 
+  async getApiReadyHealth() {
```
Comment (alibombaye) [inline app/apiProvider.js]: Change to `BuyingCatalogueApi`

File: app/pages/health/ready/getReadyStatus.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ApiProvider } from '../../../apiProvider';
+import { status } from '../status';
+
+export async function getReadyStatus() {
+  const BAPI = (await new ApiProvider().getApiReadyHealth()).data;
```
Comment (alibombaye) [inline app/pages/health/ready/getReadyStatus.js]: can we be explicit with these names

File: app/pages/health/routes.test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  });
+
+  describe('GET /ready', () => {
+    it('should return \'Healthy\' when BAPI and DAPI are both \'Healthy\'', testHealthStatus('/ready', {
+      BAPI: status.healthy,
+      DAPI: status.healthy,
+      expected: status.healthy,
+    }));
+
+    it('should return \'Degraded\' when BAPI is \'Healthy\' and DAPI is \'Degraded\'', testHealthStatus('/ready', {
```
Comment (alibombaye) [inline app/pages/health/routes.test.js]: don't think DAPI can be degraded

File: app/pages/health/live/getLiveStatus.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { status } from '../status';
```
Comment (kat1906) [inline app/pages/health/live/getLiveStatus.js]: you can add this function and getReadyStatus to the same file and they can just live in the health folder

File: app/pages/health/routes.test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+describe('GET /health', () => {
+  describe('GET /live', () => {
+    it('should return \'Healthy\' when app is running', testHealthStatus('/live', {
+      expected: status.healthy,
+    }));
+  });
+
+  describe('GET /ready', () => {
+    it('should return \'Healthy\' when BAPI and DAPI are both \'Healthy\'', testHealthStatus('/ready', {
```
Comment (kat1906) [inline app/pages/health/routes.test.js]: please change to 
testHealthStatus(
  route: '/ready',
  BAPI: status.healthy,
  DAPI: status.healthy,
  expected: status.healthy,
)
 for each test (and you will have to update the `testHealthStatus` function to accept the params this way)

File: app/pages/health/live/getLiveStatus.js (hunk 4 above)
Comment (BenBJSS) [reply app/pages/health/live/getLiveStatus.js]: I think it's nice to have them separated into a sub-directory that matches their /:route name

**Pull request merged**